src/Script/RamCalculations.ts

```diff
@@ -80,115 +91,108 @@ function parseOnlyRamCalculate(otherScripts: Map<ScriptFilePath, Script>, code:
-
-    // Process additional modules, which occurs if the "main" script has any imports
-    while (parseQueue.length > 0) {
-      const nextModule = parseQueue.shift();
-      if (nextModule === undefined) throw new Error("nextModule should not be undefined");
-      if (nextModule.startsWith("https://") || nextModule.startsWith("http://")) continue;
-
-      // Using root as the path base right now. Difficult to implement
-      const filename = resolveScriptFilePath(nextModule, root, ns1 ? ".script" : ".js");
-      if (!filename) return { cost: RamCalculationErrorCode.ImportError }; // Invalid import path
-      const script = otherScripts.get(filename);
-      if (!script) return { cost: RamCalculationErrorCode.ImportError }; // No such file on server
-
-      parseCode(script.code, nextModule);
-    }
+  // Parse the initial module, which is the "main" script that is being run
+  const initialModule = "__SPECIAL_INITIAL_MODULE__";
+  parseCode(code, initialModule);
 
-    // Finally, walk the reference map and generate a ram cost. The initial set of keys to scan
-    // are those that start with __SPECIAL_INITIAL_MODULE__.
-    let ram = RamCostConstants.Base;
-    const detailedCosts: RamUsageEntry[] = [{ type: "misc", name: "baseCost", cost: RamCostConstants.Base }];
-    const unresolvedRefs = Object.keys(dependencyMap).filter((s) => s.startsWith(initialModule));
-    const resolvedRefs = new Set();
-    const loadedFns: Record<string, boolean> = {};
-    while (unresolvedRefs.length > 0) {
-      const ref = unresolvedRefs.shift();
-      if (ref === undefined) throw new Error("ref should not be undefined");
-
-      // Check if this is one of the special keys, and add the appropriate ram cost if so.
-      if (ref === "hacknet" && !resolvedRefs.has("hacknet")) {
-        ram += RamCostConstants.HacknetNodes;
-        detailedCosts.push({ type: "ns", name: "hacknet", cost: RamCostConstants.HacknetNodes });
-      }
-      if (ref === "document" && !resolvedRefs.has("document")) {
-        ram += RamCostConstants.Dom;
-        detailedCosts.push({ type: "dom", name: "document", cost: RamCostConstants.Dom });
-      }
-      if (ref === "window" && !resolvedRefs.has("window")) {
-        ram += RamCostConstants.Dom;
-        detailedCosts.push({ type: "dom", name: "window", cost: RamCostConstants.Dom });
-      }
+  // Process additional modules, which occurs if the "main" script has any imports
+  while (parseQueue.length > 0) {
+    const nextModule = parseQueue.shift();
+    if (nextModule === undefined) throw new Error("nextModule should not be undefined");
+    if (nextModule.startsWith("https://") || nextModule.startsWith("http://")) continue;
 
-      resolvedRefs.add(ref);
+    // Using root as the path base right now. Difficult to implement
+    const filename = resolveScriptFilePath(nextModule, root, ns1 ? ".script" : ".js");
+    if (!filename) return { errorCode: RamCalculationErrorCode.ImportError }; // Invalid import path
```

Comment:
we could actually make this transparent to players too by adding fileName to `error` (and probably changing it to `errorMessage` (string | undefined). wdyt?

Reply:
I assume it totally makes sense and added it as well.